ui.apps/src/main/javascript/minicart/src/components/Minicart/product.js

```diff
@@ -0,0 +1,92 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+import React, { useState, useMemo, useCallback } from 'react';
+import { func, number, shape, object } from 'prop-types';
+import { Price } from '@magento/peregrine';
+import classes from './product.css';
+
+import { transparentPlaceholder } from '../../utils/transparentPlaceholder';
+import makeUrl from '../../utils/makeUrl';
+import Kebab from './kebab';
+import Section from './section';
+import { string } from 'postcss-selector-parser';
+
+const imageWidth = 80;
+const imageHeight = 100;
+
+const Product = props => {
+    const { beginEditItem, item, removeItemFromCart } = props;
+
+    const { product = undefined, quantity = 0, id = '' } = item;
```

Comment:
You may want to default `product` to an empty object instead of `undefined` because right below this you're attempting to pull values off of it, which will throw an exception if it is `undefined`.

Granted we're probably already in an error state if `product` gets defaulted, but at least it won't blow up on you.